crates/core/src/models/mod.rs

```diff
@@ -234,3 +239,73 @@ impl ConvertibleStarknetBlock for BlockWithTxs {
         Ok(block.into())
     }
 }
+
+pub struct StarknetEvent(Event);
+
+impl StarknetEvent {
+    pub fn new(sn_event: Event) -> Self {
+        Self(sn_event)
+    }
+}
+
+#[async_trait]
+impl ConvertibleStarknetEvent for StarknetEvent {
+    async fn to_eth_log(&self, _client: &dyn KakarotClient) -> Result<Log, KakarotClientError> {
+        // let starknet_block_id = StarknetBlockId::Tag(BlockTag::Latest);
+        // let address = match client.get_evm_address(&self.0.from_address, &starknet_block_id).await {
+        //     Ok(address) => address,
+        //     Err(_) => {
+        //         return Err(KakarotClientError::OtherError(anyhow::anyhow!(
+        //             "Starknet event address has no EVM mapping",
+        //         )));
+        //     }
+        // };
+
+        // Convert the StarkNet address to a byte array
+        let sn_bytes = self.0.from_address.clone().to_bytes_be();
+        // Take the last 20 bytes of the StarkNet address to create the Ethereum address
+        let address = Address::from_slice(&sn_bytes[12..32]);
+
+        let topics: Result<Vec<H256>, KakarotClientError> = self
+            .0
+            .keys
+            .chunks(2)
+            .map(|chunk| {
+                let low = BigUint::from_bytes_be(&chunk[0].to_bytes_be());
```

Comment:
Why are we using BigUint here?
Aren't low and high "forced" to be u128?

Could we introduce a check to see if chunk[0] > u128::max (if that exists) ?
If it is higher, then it can't be a uint256 as defined in Cairo
then we can simply use u128 all around?

Reply:
was used in response to the comment here:
https://github.com/kkrt-labs/kakarot-rpc/blob/b636837c205e1a64d399c9d9418946082b6c2044/crates/core/src/client/mod.rs#L535